x-pack/platform/plugins/shared/onechat/server/services/chat/utils/resolve_selected_connector_id.ts

```diff
@@ -0,0 +1,96 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { KibanaRequest } from '@kbn/core-http-server';
+import type { UiSettingsServiceStart } from '@kbn/core-ui-settings-server';
+import type { SavedObjectsServiceStart } from '@kbn/core-saved-objects-server';
+import type { InferenceServerStart } from '@kbn/inference-plugin/server';
+import type { InferenceConnector } from '@kbn/inference-common';
+import { InferenceConnectorType } from '@kbn/inference-common';
+import {
+  GEN_AI_SETTINGS_DEFAULT_AI_CONNECTOR,
+  GEN_AI_SETTINGS_DEFAULT_AI_CONNECTOR_DEFAULT_ONLY,
+} from '@kbn/management-settings-ids';
+
+// TODO: Import from gen-ai-settings-plugin (package) once available
+const NO_DEFAULT_CONNECTOR = 'NO_DEFAULT_CONNECTOR';
+
+const selectDefaultConnector = ({ connectors }: { connectors: InferenceConnector[] }) => {
+  const inferenceConnector = connectors.find(
+    (connector) => connector.type === InferenceConnectorType.Inference
+  );
+  if (inferenceConnector) return inferenceConnector;
+
+  const openAIConnector = connectors.find(
+    (connector) => connector.type === InferenceConnectorType.OpenAI
+  );
+  if (openAIConnector) return openAIConnector;
+
+  return connectors[0];
+};
+
+const tryGetInferenceDefault = async (
+  inference: InferenceServerStart,
+  request: KibanaRequest
+): Promise<string | undefined> => {
+  try {
+    const defaultConnector = await inference.getDefaultConnector(request);
+    return defaultConnector.connectorId;
+  } catch {
+    return undefined;
+  }
+};
+
+const tryGetFallbackConnector = async (
+  inference: InferenceServerStart,
+  request: KibanaRequest
+): Promise<string | undefined> => {
+  try {
+    const connectors = await inference.getConnectorList(request);
+    if (connectors.length > 0) {
+      const fallbackConnector = selectDefaultConnector({ connectors });
+      return fallbackConnector.connectorId;
+    }
+  } catch {
+    // Ignore errors
+  }
+  return undefined;
+};
+
+export const resolveSelectedConnectorId = async ({
+  uiSettings,
+  savedObjects,
+  request,
+  connectorId,
+  inference,
+}: {
+  uiSettings: UiSettingsServiceStart;
+  savedObjects: SavedObjectsServiceStart;
+  request: KibanaRequest;
+  connectorId?: string;
+  inference: InferenceServerStart;
+}): Promise<string | undefined> => {
+  const soClient = savedObjects.getScopedClient(request);
+  const uiSettingsClient = uiSettings.asScopedToClient(soClient);
+
+  const [defaultConnectorSetting, defaultConnectorOnly] = await Promise.all([
+    uiSettingsClient.get<string>(GEN_AI_SETTINGS_DEFAULT_AI_CONNECTOR),
+    uiSettingsClient.get<boolean>(GEN_AI_SETTINGS_DEFAULT_AI_CONNECTOR_DEFAULT_ONLY),
+  ]);
+
+  const hasValidDefaultConnector =
+    defaultConnectorSetting && defaultConnectorSetting !== NO_DEFAULT_CONNECTOR;
+
+  if (defaultConnectorOnly && hasValidDefaultConnector) return defaultConnectorSetting;
+  if (connectorId) return connectorId;
```

Comment:
So, I'm not 100% sure about this?

The way I see it, from the UI if that value is set, we can only "pick" that default connector (e.g from dropdown lists in the connector selector widget in the assistants and onechat).

But does `defaultConnectorOnly` also mean we're restraining the `connectorId` parameter when calling the converse API directly (via http), or is it meant to be a "UI thing"? 

If it's the former, we may want to throw an error if `defaultConnectorOnly=true` and `connectorId` is specified and `connectorId !==defaultConnectorSetting`? May be more explicit than silently falling back to another connector than the one explicitly provided when calling the API?

Not blocker - we can easily revisit later, but genuinely wondering.